### ISSUE TITLE
A whole tree is omitted from the result depending on a query

### ISSUE THREAD

**Issue by sungwoncho** [dgraph-io/dgraph#1023]
```
Dgraph version   : v0.7.7
Commit SHA-1     : e5445ee
Commit timestamp : 2017-06-06 12:23:45 +1000
Branch           : HEAD
```

---

**Reproduction**

1. Load the following dataset

[dgraph-1-2017-06-07-12-47.rdf.gz](https://github.com/dgraph-io/dgraph/files/1056782/dgraph-1-2017-06-07-12-47.rdf.gz)
[dgraph-schema-1-2017-06-07-12-47.rdf.gz](https://github.com/dgraph-io/dgraph/files/1056783/dgraph-schema-1-2017-06-07-12-47.rdf.gz)

2. Run the following query

```
{
    question(id: 0x1e2b04df0926d920) {
      _uid_
      Title {
        Text
      }
      Body {
        Text
      }
      questionTags as Tags {
        TagName: Text
      }
    }

    tags(id: var(questionTags)) {
      relatedQuestions: ~Tags(first: 10) {
        _uid_
        Title {
          Text
        }
        UpvoteCount: count(Upvote)
        DownvoteCount: count(Downvote)
      }
    }
  }
```

Observe that the response is in the shape of:

```
{
  question: [ ... ],
  tags: [ ... ]
}
```

3. Run the following query

```
{
    question(id: 0x1e2b04df0926d920) {
      _uid_
      Title {
        Text
      }
      Body {
        Text
      }
      questionTags as Tags {
        TagName: Text
      }
    
      Has.Answer {
        Body {
          Text
        }
      }
    }

    tags(id: var(questionTags)) {
      relatedQuestions: ~Tags(first: 10) {
        _uid_
        Title {
          Text
        }
        UpvoteCount: count(Upvote)
        DownvoteCount: count(Downvote)
      }
    }
  }
```

(same query except we added `Has.Answer` block to `question`.

Now `question` is omitted from the result.

```
{
  tags: [ ... ]
}
```

**Comment by pawanrawal** [comment]
This is because of the automatic cascading that we do when a block has a variable defined but doesn't use it. 

We will get rid of it because the user now has the option to specify the `@cascade` directive if they want to. Pushing a fix.

**Comment by pawanrawal** [comment]
Fix merged to `master` and `release/v0.7.7`. Binaries should be updated in an hour.
